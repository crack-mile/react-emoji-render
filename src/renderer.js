import React, { PropTypes } from "react";
import classnames from "classnames";
import isString from "lodash.isstring";
import replace from "string-replace-to-array";
import emojiRegex from "emoji-regex";
import asciiRegex from "./asciiRegex";
import normalizeProtocol from "./normalizeProtocol";
import unicodeToCodepoint from "./unicodeToCodepoint";
import aliases from "../data/aliases";
import asciiAliases from "../data/asciiAliases";

const asciiAliasesRegex = asciiRegex();
const unicodeEmojiRegex = emojiRegex();
const aliasesRegex = /:([\w\-\_]+):/g;

// using em's we can ensure size matches surrounding font
const style = {
  width: "1em",
  height: "1em",
  margin: "0 .05em 0 .1em",
  verticalAlign: "-0.1em"
};

export default function Emoji(
  { text, onlyEmojiClassName, options = {}, className, ...rest }
) {
  const protocol = normalizeProtocol(options.protocol);

  function replaceUnicodeEmoji(match, i) {
    if (!options.baseUrl) {
      return (
        <span key={i} style={style} className={options.className}>{match}</span>
      );
    }

    const separator = options.size ? "/" : "";
    const codepoint = unicodeToCodepoint(match);
    const src = `${protocol}${options.baseUrl}${options.size}${separator}${codepoint}.${options.ext}`;

    return (
      <img
        key={i}
        src={src}
        style={style}
        className={options.className}
        {...options.props}
      />
    );
  }

  function replaceAsciiAliases(...match) {
    const asciiAliasKeys = Object.keys(asciiAliases);

    for (let i in asciiAliasKeys) {
      const alias = asciiAliasKeys[i];
      const data = asciiAliases[alias];
      if (data.includes(match[2])) {
        return `:${alias}:`;
      }
    }
    return match;
  }

  function replaceAliases(...match) {
    return aliases[match[1]] || match[0];
  }

  function isOnlyEmoji(output) {
    if (output.length > 3) return false;

    for (let i = 0; i < output.length; i++) {
      if (typeof output[i] === "string") return false;
    }

    return true;
  }

  const output = replace(
    text
      .replace(asciiAliasesRegex, replaceAsciiAliases)
      .replace(aliasesRegex, replaceAliases),
    unicodeEmojiRegex,
    replaceUnicodeEmoji
  );

  const classes = classnames(className, {
    [onlyEmojiClassName]: isOnlyEmoji(output)
  });

  return (
    <span {...rest} classes={classes}>
      {output}
    </span>
  );
}

Emoji.propTypes = {
  text: PropTypes.string,
  props: PropTypes.object,
  onlyEmojiClassName: PropTypes.string,
  options: PropTypes.shape({
    baseUrl: PropTypes.string,
    size: PropTypes.string,
    ext: PropTypes.string,
    className: PropTypes.string
  })
};
