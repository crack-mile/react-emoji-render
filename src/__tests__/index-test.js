import React from "react";
import Emoji from "../../src/index";
import renderer from "react-test-renderer";

test("Handles strings with no emoji", () => {
  const component = renderer.create(<Emoji>Facebook</Emoji>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Handles emoji with a single codepoint", () => {
  const component = renderer.create(<Emoji>This ❤️ is 👌</Emoji>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Handles emoji with a multiple codepoints", () => {
  const component = renderer.create(<Emoji>Great work 👍🏾 👨‍👩‍👧‍👦</Emoji>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
