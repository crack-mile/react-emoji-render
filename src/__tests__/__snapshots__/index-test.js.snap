exports[`Emoji a mixture of emoji syntax 1`] = `
<span
  classes="">
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    😆
  </span>
   This is a selection of 
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    💩
  </span>
   emoji
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    😃
  </span>
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    👌🏿
  </span>
</span>
`;

exports[`Emoji aliases containing underscores 1`] = `
<span
  classes="">
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    😜
  </span>
   wow
</span>
`;

exports[`Emoji aliases with skin tone modifiers 1`] = `
<span
  classes="">
  Say hello to 
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    👩🏿
  </span>
</span>
`;

exports[`Emoji ascii aliases 1`] = `
<span
  classes="">
  That\'s awesome
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    😃
  </span>
</span>
`;

exports[`Emoji does nothing to unknown aliases 1`] = `
<span
  classes="">
  An :unknown: alias
</span>
`;

exports[`Emoji emoji with a multiple codepoints 1`] = `
<span
  classes="">
  Great work 
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    👍🏾
  </span>
   
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    👨‍👩‍👧‍👦
  </span>
</span>
`;

exports[`Emoji emoji with a single codepoint 1`] = `
<span
  classes="">
  This 
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    ❤️
  </span>
   is 
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    👌
  </span>
</span>
`;

exports[`Emoji four emoji should not add onlyEmojiClassName 1`] = `
<span
  classes="">
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    👋
  </span>
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    😀
  </span>
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    👍🏾
  </span>
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    💞
  </span>
</span>
`;

exports[`Emoji just emoji should add onlyEmojiClassName 1`] = `
<span
  classes="onlyEmojiClass">
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    😀
  </span>
</span>
`;

exports[`Emoji just emoticon should add onlyEmojiClassName 1`] = `
<span
  classes="onlyEmojiClass">
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    😄
  </span>
</span>
`;

exports[`Emoji simple aliases 1`] = `
<span
  classes="">
  This 
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    😄
  </span>
   is nice 
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    💩
  </span>
</span>
`;

exports[`Emoji strings with no emoji 1`] = `
<span
  classes="">
  Just some words
</span>
`;

exports[`Emoji three emoji should add onlyEmojiClassName 1`] = `
<span
  classes="onlyEmojiClass">
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    😀
  </span>
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    👍🏾
  </span>
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    💞
  </span>
</span>
`;

exports[`Emoji with svg prop 1`] = `
<span
  classes=""
  svg={true}>
  This 
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    👨🏿
  </span>
   is 
  <span
    className={undefined}
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    }>
    👌
  </span>
</span>
`;

exports[`Emojione a mixture of emoji syntax 1`] = `
<span
  classes="">
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f606.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
   This is a selection of 
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f4a9.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
   emoji
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f603.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f44c-1f3ff.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Emojione aliases containing underscores 1`] = `
<span
  classes="">
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f61c.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
   wow
</span>
`;

exports[`Emojione aliases with skin tone modifiers 1`] = `
<span
  classes="">
  Say hello to 
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f469-1f3ff.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Emojione ascii aliases 1`] = `
<span
  classes="">
  That\'s awesome
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f603.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Emojione does nothing to unknown aliases 1`] = `
<span
  classes="">
  An :unknown: alias
</span>
`;

exports[`Emojione emoji with a multiple codepoints 1`] = `
<span
  classes="">
  Great work 
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f44d-1f3fe.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
   
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f468-200d-1f469-200d-1f467-200d-1f466.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Emojione emoji with a single codepoint 1`] = `
<span
  classes="">
  This 
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/2764.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
   is 
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f44c.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Emojione four emoji should not add onlyEmojiClassName 1`] = `
<span
  classes="">
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f44b.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f600.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f44d-1f3fe.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f49e.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Emojione just emoji should add onlyEmojiClassName 1`] = `
<span
  classes="onlyEmojiClass">
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f600.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Emojione just emoticon should add onlyEmojiClassName 1`] = `
<span
  classes="onlyEmojiClass">
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f604.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Emojione simple aliases 1`] = `
<span
  classes="">
  This 
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f604.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
   is nice 
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f4a9.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Emojione strings with no emoji 1`] = `
<span
  classes="">
  Just some words
</span>
`;

exports[`Emojione three emoji should add onlyEmojiClassName 1`] = `
<span
  classes="onlyEmojiClass">
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f600.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f44d-1f3fe.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/png/1f49e.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Emojione with svg prop 1`] = `
<span
  classes="">
  This 
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/svg/1f468-1f3ff.svg"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
   is 
  <img
    className={undefined}
    src="http://cdnjs.cloudflare.com/ajax/libs/emojione/2.2.7/assets/svg/1f44c.svg"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Twemoji a mixture of emoji syntax 1`] = `
<span
  classes="">
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f606.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
   This is a selection of 
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f4a9.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
   emoji
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f603.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f44c-1f3ff.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Twemoji aliases containing underscores 1`] = `
<span
  classes="">
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f61c.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
   wow
</span>
`;

exports[`Twemoji aliases with skin tone modifiers 1`] = `
<span
  classes="">
  Say hello to 
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f469-1f3ff.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Twemoji ascii aliases 1`] = `
<span
  classes="">
  That\'s awesome
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f603.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Twemoji does nothing to unknown aliases 1`] = `
<span
  classes="">
  An :unknown: alias
</span>
`;

exports[`Twemoji emoji with a multiple codepoints 1`] = `
<span
  classes="">
  Great work 
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f44d-1f3fe.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
   
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f468-200d-1f469-200d-1f467-200d-1f466.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Twemoji emoji with a single codepoint 1`] = `
<span
  classes="">
  This 
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/2764.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
   is 
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f44c.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Twemoji four emoji should not add onlyEmojiClassName 1`] = `
<span
  classes="">
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f44b.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f600.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f44d-1f3fe.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f49e.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Twemoji just emoji should add onlyEmojiClassName 1`] = `
<span
  classes="onlyEmojiClass">
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f600.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Twemoji just emoticon should add onlyEmojiClassName 1`] = `
<span
  classes="onlyEmojiClass">
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f604.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Twemoji simple aliases 1`] = `
<span
  classes="">
  This 
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f604.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
   is nice 
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f4a9.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Twemoji strings with no emoji 1`] = `
<span
  classes="">
  Just some words
</span>
`;

exports[`Twemoji three emoji should add onlyEmojiClassName 1`] = `
<span
  classes="onlyEmojiClass">
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f600.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f44d-1f3fe.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/72x72/1f49e.png"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;

exports[`Twemoji with svg prop 1`] = `
<span
  classes="">
  This 
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/svg/1f468-1f3ff.svg"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
   is 
  <img
    className={undefined}
    src="http://twemoji.maxcdn.com/2/svg/1f44c.svg"
    style={
      Object {
        "height": "1em",
        "margin": "0 .05em 0 .1em",
        "verticalAlign": "-0.1em",
        "width": "1em",
      }
    } />
</span>
`;
